File: 2020 czerwiec/c++/2.cpp
#include <iostream>
#include <fstream>
#include <string>
using namespace std;


int main()
{
    ifstream in("pary.txt");

    int a;
    string s;
    while (in >> a >> s)
    {
        pair <int, char> ans = {1, s[0]};
        char prv = ' ';
        int cur = 0;
        for (char c: s)
        {
            if (c == prv)
            {
                cur++;
                if (cur > ans.first)
                    ans = {cur, c};
            }
            else cur = 1;

            prv = c;
        }

        for (int i = 0; i < ans.first; i++)
            cout << ans.second;
        cout << ' ' << ans.first << '\n';
    }
}

File: 2020 czerwiec/python/2.py
with open("pary.txt", "r") as f:
    pary = f.read().strip().split('\n')

for p in pary:
    n, s = p.split()
    best, symbol, prv, cur = 1, s[0], ' ', 0
    for c in s:
        if c == prv:
            cur += 1
            if cur > best:
                best = cur
                symbol = c
        else:
            cur = 1
        
        prv = c
    
    print(symbol * best, best)